### PR TITLE
ci: CI Workflow Optimization

### PR DESCRIPTION
**Pull request by tippmar-nr**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

* Refactors our current "All Solutions Build" CI workflow as follows:
    * Removes the profiler build job (refactored to a separate workflow, more on that below.
    * Removes the Unit Test execution step from the full agent build job.

* Adds a new workflow (`build_profiler.yml`) to check for modified Profiler source files, build the Profiler if any were modified, publish the binaries in a NuGet package on MyGet.org and update the package version reference in `Home.csproj` (via a PR). This workflow is triggered by push to any branch other than `main` and `feature\**`, and has a `workflow_dispatch` trigger allowing it to be manually invoked on any branch.
* Removes profiler binaries from the repo.
* Modifies the `build_home.ps1` script to copy theProfiler binaries from the `Home.csproj` output folder rather than from the `_profilerBuild` folder. 

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteService.cs [hunk 1]
```diff
@@ -114,6 +114,9 @@ private void PublishWithDotnetExe(string framework)
 
             TestLogger?.WriteLine($"[RemoteService]: Publishing to {deployPath}.");
 
+            var sw = new Stopwatch();
```
Comment (nrcventura) [inline tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteService.cs:117]: Is this supposed to be part of this PR?

File: .github/workflows/build_profiler.yml [hunk 2]
```diff
@@ -1,42 +1,361 @@
...
+  check-for-changes:
+    name: Check for Updated Profiler Files
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    outputs:
+      profiler_src: ${{ steps.filter.outputs.profiler_src }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@03bee3930647ebbf994244c21ddbc0d4933aab4f # v2.3.0
```
Comment (nrcventura) [inline .github/workflows/build_profiler.yml:50]: I think this needs the latest harden runner updates.

File: tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteService.cs (hunk 1 above)
Comment (tippmar-nr) [reply tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteService.cs:117]: Yeah, I threw it in when trying to investigate where time was being spent, it seemed like the publish step was slow sometimes.

File: .github/workflows/build_profiler.yml (hunk 2 above)
Comment (tippmar-nr) [reply .github/workflows/build_profiler.yml:50]: The change got picked up when I merged from `main`.

**Review by jaffinito — APPROVED** [review]
LGTM

File: .github/workflows/build_profiler.yml (hunk 2 above)
Comment (nrcventura) [reply .github/workflows/build_profiler.yml:50]: It looks like it didn't pick up the updates in this workflow. The All Solutions workflow is on 2.4.0, but this one is still on 2.3.0.

**Review by nrcventura — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Pull request merged**